src/Microsoft.AspNet.Mvc.TagHelpers/InputTagHelper.cs

```diff
@@ -249,12 +249,12 @@ private void GenerateCheckBox(ModelExplorer modelExplorer, TagHelperOutput outpu
                 output.Attributes.Clear();
                 output.TagName = null;
 
-                output.Content.Append(tagBuilder.ToString(TagRenderMode.SelfClosing));
+                output.Content.Append(tagBuilder.ToHtmlContent(TagRenderMode.SelfClosing));
```

Comment:
@rynowak if `TagBuilder` implemented `IHtmlContent`, would have to decide on the `TagRenderMode` it would use when called that way and would still need `ToHtmlContent` for the other `TagRenderMode` choices. Probably better to leave as @sornaks has it here.